src/components/SQForm/__tests__/SQFormDropdown.test.js

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import * as Yup from 'yup';
+import {screen} from '@testing-library/react';
+import SQFormDropdown from '../SQFormDropdown';
+import {renderSQForm} from '../../../utils/tests';
+import userEvent from '@testing-library/user-event';
+
+const DEFAULT_PROPS = {
+  name: 'dropdownExample',
+  label: 'Dropdown Example'
+};
+
+const DEFAULT_INTITIAL_VALUES = {
+  initialValues: {
+    dropdownExample: ''
+  }
+};
+
+const DEFAULT_OPTIONS = [
+  {label: 'Test 1', value: 1},
+  {label: 'Test 2', value: 2},
+  {label: 'Test 3', value: 3}
+];
+
+const createDropdown = (
+  options = DEFAULT_OPTIONS,
+  fieldProps = DEFAULT_PROPS
+) => {
+  return <SQFormDropdown {...fieldProps}>{options}</SQFormDropdown>;
+};
+
+it('should render list of options', () => {
+  const renderDropdown = createDropdown();
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const optionsList = screen.getByRole('listbox', /dropdown example/i);
+
+  expect(optionsList).toBeInTheDocument();
+  expect(optionsList.children.length).toBe(3);
+});
+
+it('should render with initial value', () => {
+  const initialValues = {
+    dropdownExample: 2
+  };
+
+  const renderDropdown = createDropdown();
+
+  renderSQForm(renderDropdown, {initialValues});
+
+  const labelValue = screen.getByRole('button', /dropdown example/i);
+  expect(labelValue).toHaveTextContent('Test 2');
+});
+
+it('should update when option is selected', () => {
+  const handleChange = jest.fn();
+  const fieldProps = {
+    ...DEFAULT_PROPS,
+    onChange: handleChange
+  };
+
+  const renderDropdown = createDropdown(DEFAULT_OPTIONS, fieldProps);
+
+  renderSQForm(renderDropdown, DEFAULT_INTITIAL_VALUES);
+
+  const expandButton = screen.getByRole('button', /dropdown example/i);
+  userEvent.click(expandButton);
+
+  const option = screen.getByText('Test 1');
+  userEvent.click(option);
+
+  expect(handleChange).toHaveBeenCalled();
+  expect(handleChange).toHaveBeenCalledTimes(1);
+
+  const labelValue = screen.getByRole('button', /dropdown example/i);
+  expect(labelValue).toHaveTextContent('Test 1');
```

Comment:
**suggestion**: A value check before making the change would give more confidence that this test is working as intended.
e.g. `expect(labelValue).not.toHaveTextContent('Test 1')`

Also **nitpick**: I think we should use something like `DEFAULT_OPTIONS[0].label` for these values because if later on someone reorders/changes that object it'll cause this test to fail when it really shouldn't